---
title: "Class 13: DESeq Analysis"
author: "Arshiya Zarmahd (PID: A16247996)
format: pdf
---

This week we are looking at differential expression analysis. 

The data for this hands-on session comes from a published RNA-seq experiment where airway smooth muscle cells were treated with dexamethasone, a synthetic glucocorticoid steroid with anti-inflammatory effects (Himes et al., 2014). 


































## Add Annotation Data

We want to add on gene symbols (ie. gene names) as well as other common identifiers from major databses for all our genes of interest. 

```{r}
library(AnnotationDbi)
library(org.Hs.eg.db)
```

```{r}
columns(org.Hs.eg.db)
```

```{r}
head(res)
```

My IDs are in the `rownames(res)` and they are from ENEMBL.

```{r}
res$symbol <- mapIds(org.Hs.eg,
                    keys=rownames(res),
                    keytype="ENSEMBL",
                    column="SYMBOL",
                    multiVals="first")
head(res)
```

We also want "GENENAME" and "ENTREZID" 

```{r}
res$genename <- mapIds(org.Hs.eg,
                    keys=rownames(res),
                    keytype="ENSEMBL",
                    column="GENENAME",
                    multiVals="first")
res$entrez <- mapIds(org.Hs.eg,
                    keys=rownames(res),
                    keytype="ENSEMBL",
                    column="ENTREZID",
                    multiVals="first")
head(res)
```

Lets save our results to a new CSV file

```{r}
write.csv(res, file="myresults.csv")
```

## Pathway Analysis

Here we will use the "gage" package to do some pathway analysis (aka, geneset enrichment).

```{r}
library(pathview)
library(gage)
library(gageData)
```

Have a wee peak at KEGG data.

```{r}
data(kegg.sets.hs)

head(kegg.sets.hs)
```

To run gage, we need to provide it with a vector of fold-change valyes (not our big full results table).

```{r}
folchanges <- res$log2FoldChange
#foldchanges
```

Add the ENTREZ ids as names to this vector/

```{r}
c(chandra=10, alice=9, barry=7)
names(x) <- c("chandra", "alice", "barry")
```

Add ENTREZ ids as names to my foldchanges vector.

```{r}
names(foldchanges) <- res$entrez
head(foldchanges)
```

Now run **gage** with this input and KEGG pathways.

```{r}
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```

```{r}
attributes(keggres)
```

```{r}
head(keggres$less)
```

Lets have a look at the hsa05310 Asthma pathway with our genes highlighted using the `pathview()` function.

```{r}
pathview(gene.data=foldchanges, pathway.id="hsa05310")
```

![](hsa05310.pathview.png)







